src/pool/pool_disjoint.c

```diff
@@ -646,18 +646,26 @@ static void free_slab(void *unused, void *slab) {
 
 umf_result_t disjoint_pool_initialize(umf_memory_provider_handle_t provider,
                                       const void *params, void **ppPool) {
-    // TODO set defaults when user pass the NULL as params
-    if (!provider || !params || !ppPool) {
+    if (!provider || !ppPool) {
         return UMF_RESULT_ERROR_INVALID_ARGUMENT;
     }
 
-    const umf_disjoint_pool_params_t *dp_params = params;
+    const umf_disjoint_pool_params_t *dp_params;
 
-    // min_bucket_size parameter must be a power of 2 for bucket sizes
-    // to generate correctly.
-    if (!dp_params->min_bucket_size ||
-        !IS_POWER_OF_2(dp_params->min_bucket_size)) {
-        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    // If params is NULL, use default values
+    umf_disjoint_pool_params_t default_params = {
+        .slab_min_size = 64 * 1024,           // 64KB default
+        .max_poolable_size = 2 * 1024 * 1024, // 2MB default
+        .capacity = 4,                        // default
+        .min_bucket_size = 8,                 // default
+        .cur_pool_size = 0,
+        .pool_trace = 0,
+        .shared_limits = NULL,
+        .name = "disjoint"};
+    if (!params) {
+        dp_params = &default_params;
+    } else {
+        dp_params = params;
```

Comment:
you should just create params structure with default values. Now we have default values in two places, so if we change default we must remember to change them here too